src/ui/SessionExpiryTracker/SessionExpiryTracker.tsx

```diff
@@ -0,0 +1,75 @@
+import Cookies from 'js-cookie'
+import React, { useCallback, useEffect, useState } from 'react'
+import { Redirect } from 'react-router-dom'
+
+import { useNavLinks } from 'services/navigation'
+
+const ONE_MINUTE_MILLIS = 60 * 1000
+const TWO_MINUTES_MILLIS = 2 * ONE_MINUTE_MILLIS
+const THIRTY_MINUTES_MILLIS = 30 * ONE_MINUTE_MILLIS
+const LOCAL_STORAGE_SESSION_EXPIRED_KEY = 'expired-session'
+
+const SessionExpiryTracker: React.FC = () => {
+  const [redirectToLogout, setRedirectToLogout] = useState(false)
+  const sessionExpiryTimeString = Cookies.get('session_expiry')
+  localStorage.removeItem(LOCAL_STORAGE_SESSION_EXPIRED_KEY)
+  const getCheckDelay = (sessionExpiryTime: Date) => {
+    const timeLeft = sessionExpiryTime.getTime() - new Date().getTime()
+    return timeLeft > THIRTY_MINUTES_MILLIS
+      ? timeLeft - THIRTY_MINUTES_MILLIS
+      : 0
+  }
+
+  const checkSession = useCallback(
+    (sessionExpiryTime: Date) => {
+      const currentTime = new Date()
+      const timeLeft = sessionExpiryTime.getTime() - currentTime.getTime()
+      if (timeLeft <= TWO_MINUTES_MILLIS && !redirectToLogout) {
+        setRedirectToLogout(true)
+      }
+    },
+    [redirectToLogout]
+  )
+
+  const setupSessionIntervalCheck = useCallback(
+    (sessionExpiryTime: Date) => {
+      return window.setInterval(
+        () => checkSession(sessionExpiryTime),
+        ONE_MINUTE_MILLIS
+      )
+    },
+    [checkSession]
+  )
+
+  useEffect(() => {
+    if (!sessionExpiryTimeString) {
+      return
+    }
+    const sessionExpiryTime = new Date()
+
+    let intervalId: number
+    const delayBeforeStart = getCheckDelay(sessionExpiryTime)
+    const timeoutId = setTimeout(() => {
+      intervalId = setupSessionIntervalCheck(sessionExpiryTime)
+      checkSession(sessionExpiryTime)
+    }, delayBeforeStart)
+
+    return () => {
+      clearTimeout(timeoutId)
+      clearInterval(intervalId)
```

Comment:
Not really sure what's happening with some of these annotations as I'm fairly sure I've ensured these lines have been covered in the spec file.

Reply:
Hrrm, sometimes this is an issue with GitHub caching, but looking in Codecov, it does say these lines are missing 🤔

Reply:
Okay there seems to be an issue with our test splitting, not including all of the UI files

Reply:
I've pushed a fix for this in #2418